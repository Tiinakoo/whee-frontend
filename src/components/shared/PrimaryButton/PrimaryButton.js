import React from "react";
import Button from "../../private/Button/Button";
import BodyText from "../BodyText/BodyText";
import { observer } from "mobx-react-lite";

const PrimaryButton = ({ onClick, size, children, ...props }) => (
  <Button onClick={onClick} variant="primary" size={size} attention {...props}>
    <BodyText font="body" size="md">
      {children}
    </BodyText>
  </Button>
);

export default observer(PrimaryButton);
