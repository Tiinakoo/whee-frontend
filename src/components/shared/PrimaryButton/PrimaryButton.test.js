import { mount } from "enzyme";
import React from "react";

import PrimaryButton from "./PrimaryButton";

describe("PrimaryButton", () => {
  it("renders", () => {
    const component = mount(<PrimaryButton>Some button</PrimaryButton>);

    expect(component).toMatchSnapshot();
  });
});
