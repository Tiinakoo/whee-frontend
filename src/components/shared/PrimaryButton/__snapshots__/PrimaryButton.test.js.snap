// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders 1`] = `
<PrimaryButton>
  <Button
    attention={true}
    variant="primary"
  >
    <button
      className="button button__primary button__attention button__size"
      type="button"
    >
      <BodyText
        font="body"
        size="md"
      >
        <p
          className="bodyText bodyText__fontBody bodyText__sizeMd"
        >
          Some button
        </p>
      </BodyText>
    </button>
  </Button>
</PrimaryButton>
`;
